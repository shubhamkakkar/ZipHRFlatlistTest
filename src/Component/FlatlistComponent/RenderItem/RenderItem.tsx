import React from 'react';
import { ListRenderItem, View } from 'react-native';

type TRenderItem = {
    item: any
}
export default function RenderItem({ item }: TRenderItem) {
  return (
    <View />
  );
}
