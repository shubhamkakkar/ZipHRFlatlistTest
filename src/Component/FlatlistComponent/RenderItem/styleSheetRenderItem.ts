import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  card: {
    flex: 1,
    margin: 10,
    padding: 10,
  },
  cardDescription: {
    marginHorizontal: 20,
    marginTop: 20,
  },
});

export default styles;
