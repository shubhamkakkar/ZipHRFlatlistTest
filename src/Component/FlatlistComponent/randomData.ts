export type TRandomData = {
    actionTaken: string;
    appName: string;
    meetingType: string;
}

export const randomData:TRandomData[] = [
  {
    actionTaken: 'Sold Premium Base-1',
    appName: 'Vishal Mega Marts',
    meetingType: 'Sell',
  },
  {
    actionTaken: 'Sold Premium Base-2',
    appName: 'Mega Marts',
    meetingType: 'Sell',
  },
  {
    actionTaken: 'Demo Premium Base-1',
    appName: 'McD',
    meetingType: 'Demo',
  },
  {
    actionTaken: 'Demo Premium Base-2',
    appName: 'Domino Center',
    meetingType: 'Demo',
  },
  {
    actionTaken: 'Sold Premium Base-1',
    appName: 'Wall Mart',
    meetingType: 'Sell',
  },
  {
    actionTaken: 'Demo Freemium Base-1',
    appName: 'Metro Origins',
    meetingType: 'Demo',
  },
  {
    actionTaken: 'Sold Premium Base-2',
    appName: 'Single Malt',
    meetingType: 'Sell',
  },
];
