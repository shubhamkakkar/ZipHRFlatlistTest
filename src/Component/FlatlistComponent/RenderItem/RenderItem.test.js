import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import RenderItem from './RenderItem';
import { randomData } from '../randomData';

it('RenderItem matches snapshot - props: {{ item }}', () => {
  const tree = renderer.create(<RenderItem {...{ item: randomData[0] }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('RenderItem matches snapshot - props: {{ item: [] }}', () => {
  const tree = renderer.create(<RenderItem {...{ item: [] }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
