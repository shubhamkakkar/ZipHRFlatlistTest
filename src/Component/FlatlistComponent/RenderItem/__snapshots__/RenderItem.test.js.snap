// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderItem matches snapshot - props: {{ item }} 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#7b2cbf80",
      "borderRadius": 5,
      "borderWidth": 0.5,
      "flex": 1,
      "margin": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#7b2cbf80",
        "borderBottomWidth": 1,
        "marginHorizontal": 10,
        "paddingBottom": 5,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
            },
            Object {
              "color": "#7b2cbf",
              "textAlign": "left",
            },
          ]
        }
      >
        Live Mega Marts
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 20,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 15,
              "marginHorizontal": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 14,
                "lineHeight": 19,
              },
            ]
          }
        >
          Meeting Type
        </Text>
        <Text
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "fontSize": 16,
                "lineHeight": 19,
                "marginTop": 5,
              },
            ]
          }
        >
          Sell
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 15,
              "marginHorizontal": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 14,
                "lineHeight": 19,
              },
            ]
          }
        >
          Action Taken
        </Text>
        <Text
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "fontSize": 16,
                "lineHeight": 19,
                "marginTop": 5,
              },
            ]
          }
        >
          Sold Base-1
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 15,
              "marginHorizontal": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 14,
                "lineHeight": 19,
              },
            ]
          }
        >
          Phone Number
        </Text>
        <Text
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "fontSize": 16,
                "lineHeight": 19,
                "marginTop": 5,
              },
            ]
          }
        >
          9818163250
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`RenderItem matches snapshot - props: {{ item: [] }} 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#7b2cbf80",
      "borderRadius": 5,
      "borderWidth": 0.5,
      "flex": 1,
      "margin": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#7b2cbf80",
        "borderBottomWidth": 1,
        "marginHorizontal": 10,
        "paddingBottom": 5,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
            },
            Object {
              "color": "#7b2cbf",
              "textAlign": "left",
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 20,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 15,
              "marginHorizontal": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 14,
                "lineHeight": 19,
              },
            ]
          }
        >
          Meeting Type
        </Text>
        <Text
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "fontSize": 16,
                "lineHeight": 19,
                "marginTop": 5,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 15,
              "marginHorizontal": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 14,
                "lineHeight": 19,
              },
            ]
          }
        >
          Action Taken
        </Text>
        <Text
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "fontSize": 16,
                "lineHeight": 19,
                "marginTop": 5,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 15,
              "marginHorizontal": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 14,
                "lineHeight": 19,
              },
            ]
          }
        >
          Phone Number
        </Text>
        <Text
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "fontSize": 16,
                "lineHeight": 19,
                "marginTop": 5,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
